### DISCUSSION TITLE
proxy local mode 使用姿势的一些疑问

### DISCUSSION THREAD

**Discussion by VictoryAnn** [apache/rocketmq#8931] [Q&A]
当前我部署两个分片，broker-a broker-b 用proxy local启动的，两个broker分别存储不同的topic，客户端配置了两个proxy的地址，但是会出现获取topicRoute的报错。是我使用方式存在问题吗？
具体的见 https://github.com/apache/rocketmq/issues/8904

**Comment by VictoryAnn** [comment]
关于local proxy有几个疑问：
1. 多个分片的时候，客户端连接到proxy，如果topic不在当前proxy中，会转移到另外一个proxy中吗？
2. 抛开性能的区别，local proxy是否是为了支持5.x的新功能（LMQ，static topic等）的grpc接口而存在的？
3. 是否有proxy的性能测试工具，来看下local 和 cluster的性能差异？
4. 使用proxy的话，客户端还支持httpsite的方式访问吗？

烦请各位大佬指点一下，感激不尽！！！

**Reply by VictoryAnn** [reply]
1. 又深入分析了源码，proxy local mode 不会转发到其他切片的broker中，只能访问本地的broker。

**Reply by jaylu163** [reply]
有遇到过 发消息出现 task was cancle。或者就是 Message type is not specified不

不确定是哪块问题
